pkg/batchscheduler/scheduler_manager.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package batchscheduler
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/batchscheduler/interface"
+	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/batchscheduler/volcano"
+)
+
+type SchedulerManager interface {
+	GetScheduler(schedulerName string) (schedulerinterface.BatchScheduler, error)
+}
+
+type schedulerInitializeFunc func(config *rest.Config) (schedulerinterface.BatchScheduler, error)
+
+var schedulerContainers = map[string]schedulerInitializeFunc{
+	volcano.GetPluginName(): volcano.New,
+}
+
+func GetRegisteredNames() []string {
+	var pluginNames []string
+	for key := range schedulerContainers {
+		pluginNames = append(pluginNames, key)
+	}
+	return pluginNames
+}
+
+type SchedulerManagerImpl struct {
```

Comment:
I don't think we really need an interface for `SchedulerManager`. Also `schedulerContainers` can be a field of `SchedulerManager` instead of a package-level thing, and `GetRegisteredNames` can be a method of `SchedulerManager`.

Reply:
+1, it seems a `Factory` is enough :)

Reply:
@liyinan926 I don't have a favor we should or should not use interface here, updated in latest commits , but the package level function are kept since it's used before object initialized. see
```
fmt.Sprintf("Enable batch schedulers for pods' scheduling, the available batch schedulers are: (%s).", strings.Join(batchscheduler.GetRegisteredNames(), ",")))
```

Reply:
I think having an interface only makes sense if we expect to have more than one possible implementations for `SchedulerManager`, which doesn't seem the case. So I think using a `struct` is sufficient.